strs.go

```diff
@@ -142,6 +143,62 @@ func (db *RoseDB) MSet(args ...[]byte) error {
 	return nil
 }
 
+// MSetNX sets given keys to their respective values. MSetNX will not perform
+// any operation at all even if just a single key already exists.
+func (db *RoseDB) MSetNX(args ...[]byte) error {
+	db.strIndex.mu.Lock()
+	defer db.strIndex.mu.Unlock()
+
+	if len(args) == 0 || len(args)%2 != 0 {
+		return ErrWrongNumberOfArgs
+	}
+
+	// Firstly, check each keys whether they are exists.
+	for i := 0; i < len(args); i += 2 {
+		key := args[i]
+		val, err := db.getVal(key, String)
+		if err != nil {
+			return err
+		}
+
+		// Key exists in db. We discard the rest of the key-value pairs. It
+		// provides the atomicity of the method.
+		if val != nil {
+			return nil
+		}
+	}
+
+	var addedKeys [][]byte
```

Comment:
Using a map to keep track of all added keys, iterating over an array may be very slow.

Reply:
Yes, it has O(n) time complexity. Actually, my first idea was using `map` instead of an array, but, Go does not support `[]byte` in keys. That`s why I used an array. If you have any other ideas for it, we can discuss them.

Reply:
I did. You can review it.